File: backend/src/main/java/br/com/videolocadorapassatempo/model/TitleModel.java
package br.com.videolocadorapassatempo.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "title")
@Getter
@Setter
public class TitleModel implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequence_title")
    @SequenceGenerator(name = "sequence_title", sequenceName = "sequence_title", allocationSize = 1)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "year", nullable = false)
    private Integer year;

    @Column(name = "synopsis", nullable = false)
    private String synopsis;

    @Column(name = "category", nullable = false)
    private String category;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_director", referencedColumnName = "id", nullable = false)
    private DirectorModel directorModel;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_class", referencedColumnName = "id", nullable = false)
    private ClassModel classModel;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "title_actor",
            joinColumns = @JoinColumn(name = "id_title"),
            inverseJoinColumns = @JoinColumn(name = "id_actor"))
    private List<ActorModel> actorModel = new ArrayList<>();

}

File: frontend/src/app/shared/models/title.model.ts
import { Actor } from './actor.model';
import { BasicModel } from './basic.model';
import { Class } from './class.model';
import { Director } from './director.model';

export class Title extends BasicModel {
  constructor(
    id?: number,
    public actorsModel?: Actor[],
    public actorIds?: Actor[],
    public name?: string,
    public category?: string,
    public idClass?: number,
    public classModel?: Class,
    public idDirector?: number,
    public directorModel?: Director,
    public synopsis?: string,
    public year?: string
  ) {
    super(id);
  }
}
